### ISSUE TITLE
oban 0.7.0 doesn't supports `max_attempts: 1`

### ISSUE THREAD

**Issue by antonielcm** [sorentwo/oban#49]
Hi everybody!

In our project, we have a worker that updates the client status. We defined the `max_attempts: 1` option to avoid inconsistent caused by a retry without treatment (it was a good and cheap option at the moment).

But now that we updated to `0.7.0` version, our worker test case broke because of this change, and I didn't found any explanation about this.

I would like to know why this change was made, and how can I get around this?

Thanks for now!

**Comment by sorentwo** [comment]
The `max_attempts` option is definitely still in place, is documented and has integration tests to verify it.

Can you share some of your worker test case? Perhaps there is another issue with your usage (maybe due to the switch to perform/2)?

**Comment by antonielcm** [comment]
@sorentwo I'm sorry, I should have been even more explicit in my comment. Our problem is when we define the option `max_attempts` with the value `1`, which throws an exception. 

Also, I made a mistake in my first comment, the exception is thrown during the compilation, not during the test task. Here's the error:

```
== Compilation error in file lib/broker/identity_validation/idwall/move_to_manual_approval_worker.ex ==
** (ArgumentError) expected :max_attempts to be an integer greater than 1
    lib/oban/worker.ex:215: Oban.Worker.validate_opt!/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    expanding macro: Oban.Worker.__using__/1
    lib/broker/identity_validation/idwall/move_to_manual_approval_worker.ex:2: Broker.IdentityValidation.Idwall.MoveToManualApprovalWorker (module)
    (elixir) expanding macro: Kernel.use/2
    lib/broker/identity_validation/idwall/move_to_manual_approval_worker.ex:2: Broker.IdentityValidation.Idwall.MoveToManualApprovalWorker (module)
```

And the worker's code: 

```
defmodule Broker.IdentityValidation.Idwall.MoveToManualApprovalWorker do
  use Oban.Worker, queue: "identity_validation", max_attempts: 1

  @moduledoc false
  alias Broker.Accounts
  alias Broker.IdentityValidation.Idwall

  @impl Oban.Worker
  def perform(%{"id" => id}, _job) do
    id
    |> Accounts.get_client!()
    |> Idwall.move_to_manual_approval()
  end
end
```

(p.s: I changed from `perform/1` to `perform/2` and, apparently, I had no problem. This project has more than one worker and they're ok)

**Comment by sorentwo** [comment]
@antonielcm Thanks for the details! You were absolutely right. The compile time check expected `max_attempts` greater than 1, but it should have been greater than 0. This has been fixed with 6a4e2b6 and I'll release `0.7.1` soon.

**Comment by antonielcm** [comment]
Great!

Thank you for the support! :heart_eyes:

**Comment by sorentwo** [comment]
@antonielcm I released v0.7.1 which includes this fix and an important fix to prevent leaking advisory locks.
